client/client.go

```diff
@@ -282,25 +282,30 @@ func (client *Client) doSync(method, path string, query url.Values, headers map[
 }
 
 func (client *Client) doAsync(method, path string, query url.Values, headers map[string]string, body io.Reader) (changeID string, err error) {
+	_, changeID, err = client.doAsyncFull(method, path, query, headers, body)
+	return
+}
+
+func (client *Client) doAsyncFull(method, path string, query url.Values, headers map[string]string, body io.Reader) (result json.RawMessage, changeID string, err error) {
```

Comment:
the snapshot creation is a bit unique in that there's a bit of information created together (actually before) the change, so it needs returning as part of the async result. Until now the async result wasn't used (as the changeid is in meta, with a TODO to move it into the result since forever)